checker/check_result.go

```diff
@@ -97,6 +105,19 @@ func CreateProportionalScore(success, total int) int {
 	return int(math.Min(float64(MaxResultScore*success/total), float64(MaxResultScore)))
 }
 
+// CreateProportionalScoreWeighted creates the proportional score
+// between multiple successes over the total, but some proportions
+// are worth more.
+func CreateProportionalScoreWeighted(scores ...ProportionalScoreWeighted) int {
```

Comment:
I don't think it would get called like that, but will default to you

Reply:
Ah, no, we don't. It shouldn't get called if total is 0. I will review the test cases to make sure this is caught in case we have and unweighted version, and perhaps throw an error at the moment if total or weight is 0.